src/crtoolLocalStorage.js

```diff
@@ -0,0 +1,260 @@
+import C from "./js/business.logic/constants";
+
+const CHECK_FREQUENCY = 60e3;
+
+function redirectHome() {
+    window.location.href = C.START_PAGE_RELATIVE_URL;
+}
+
+/**
+ * A proxy for the localStorage API which periodically saves to the DB and
+ * can synchronize itself when the page loads.
+ */
+const ls = {
+    review: {},
+    dirty: false,
+
+    async init() {
+        const token = ls.getUrlParameter('token');
+        let review_id = token || localStorage.getItem('curriculumReviewId') || '';
+        if (!review_id) {
+            return redirectHome();
+        }
+
+        // If review_id found but a token is not set, update the URL
+        if (!token && review_id) {
+            const new_url = ls.updateUrlParameter(window.location.href, 'token', review_id);
+            window.history.pushState({ path: new_url }, '', new_url);
+        }
+
+        let db_review;
+        try {
+            db_review = await ls.fetchReviewFromServer(review_id);
+        } catch (e) {
+            console.error(e);
+            // Do not allow end users to set their own ID via the URL or localStorage.
+            localStorage.removeItem('crtool.' + review_id);
+            localStorage.removeItem('curriculumReviewId');
```

Comment:
Will this mean that the localStorage version of the review will get deleted if any type of error occurs while trying to load it from the server? Server timeouts, lost internet connection, etc?